### DISCUSSION TITLE
should we do few tasks\projects in parallel?

### DISCUSSION THREAD

**Discussion by atherdon** [atherdon/dash-back#24] [General]
задачи просто есть и я нехочу чтобы были "провисания" когда мы обсуждаем майлстоуны. мне кажется что понадобится несколько недель вашего времени - чтобы закрыть все проблемы, которые висят на мне. 

но непонятно как это организовать. другой вопрос - каким образом отправлять вам оплату.
если вы раскачиваете свой профиль на фрилансханте - мы можем его использовать. Но фрилансхант 2 раза кушает комиссию. с меня они едят 10%, которые мы могли бы просто тратить на код. и скорее всего вы платите какую то комиссию, когда переводите гривны в рубли.

если не съекьюрно - можем еще пару майлстоунов сделать на фрилансханте, а потом уже подумать о других вариантах.

у меня тетя живет в России и я отправляю ей на карту Сбера рубли. там просто перевод 4 тысячи минимум. использую нашу криптобиржу kuna.io

или можно криптой. я когда просматривал ваш гитхаб профиль - увидел что вы делали криптобота. можно использовать LTC, потому что у них коммиссия низкая. я просто сам получаю зарплату криптой - так и не нашел нормального варианта получать деньги из Америки.

cc @kolserdav

**Comment by kolserdav** [comment]
Мне сейчас нужно хант прокачать, поэтому я работаю по цене даже ниже той которую считал для себя минимальной ещё год назад, я даже готов комиссию на себя брать, как в stage_0

**Reply by atherdon** [reply]
я тут только могу подчинится. сам качал профиль - все понимаю.

но после того как "потерял" компанию, частично изза проектов на upwork - понял что прфиль вообще не гарантирует ничего. когда вся команда разбежалась и мы с партнером остались доделывать проекты сами и бесплатно - я думал что раз есть профайл - я сейчас быстренько найду проектик и буду с деньгами. 
в итоге мы просто его бросили и все. 

поэтому то я и стал писать статьи, рассказывая что я делаю - код на гитхабе могут и не посмотреть, а вот статьи много людей читает.

**Reply by atherdon** [reply]
Насчет проектов, я сделаю небольшое intro а потом можем подумать.


### 1. react-sendgrid-form 

закрытый репозиторий, но всегда могу вас добавить. посути это просто контактная форма(набор форм) которые позволят нам оптимизировать коммуникацию с авторами. сейчас мы используем gmail темплейты - но это очень не эффективно. выглядит это как простой проект на ant-d с несколькими формами внутри. а при нажатии отправляется определенное письмо из sendgrid. не самую последнюю версию можно посмотреть тут: https://intense-escarpment-64234.herokuapp.com/baa/promo-story

проблема: есть несколько блокеров. одним является создание простого nodejs microservice который просто будет звать метод sendGrid.send(). второй это деплой\preview и выкладывание в production. думаю это самое простое задание.

со временем я думаю мы эти формы просто интегрируем в наш dashboard - но я хочу запустить его раньше чем новую dash.



### 2. https://github.com/atherdon/markdown-to-email

я работаю над email newsletter - это один из дополнительных источников дохода компании. не много, но курочка по зернышку. отсылаем примерно 2 раза в неделю. и тк сервис который мы используем - он не идеальный - пришлось делать свою тулзу которая помогает из markdown переводить контент в html template который хорошо отображается в почтовиках.



после этого мы будем разбивать проект на куски и переносить темлпейт на реакт версию - чтобы нам же было просто. 

пример письма которое генерируется(вчера отослали): https://www.noonifications.tech/w/f01Li4pkuUgga7MajWexIg
пример папки с кодом для одной рассылки: https://github.com/atherdon/newsletters/tree/master/archive/Thematic/2021/Februrary/progressed/webiny3

Сейчас у нас есть 3 основные ветки.
- **master branch** with latest stuff(v0.9): https://github.com/atherdon/markdown-to-email

- **react branch** (v2.0): https://github.com/atherdon/markdown-to-email/tree/react-components-stage-zero

- **version with a working generator** that I'm using 2-3 times per week: https://github.com/atherdon/markdown-to-email/tree/plain-working-version


**проблема**

Мы сделали генерацию empty-content(середины письма, в которой у нас текст и картинки) а вот полную генерацию "забыли" доделать. это простой plain js сейчас. там вообще ничего сложного. нужно просто сесть, разобраться, сделать базовую версию - потом я протестирую как работает для разных кейсов и все.

получатеся что пока у меня нету полностью работающего генератора - нам приходится для каждого письма ручками копипастить верстку в середину. это и глупо и тратит время. особенно бывает такой клиент, который хочет чтото изменить внутри и приходится по нескольку раз переделывать. а генератор бы просто автоматически делал новый render и все.

У нас получилось добится хороших результатов в генерации HTML тегов, которые отвечают за wrapping нашего контента. Тоесть текст newsletter - генерируется нормально. Мемы и другие картинки работают и т.д.

Вы можете сравнить content.md(этот файл хранит в себе контент в markdown format): [https://github.com/atherdon/newsletters/blob/master/archive/Thematic/2021/January/flatfile8/content.md](https://github.com/atherdon/newsletters/blob/master/archive/Thematic/2021/January/flatfile8/content.md) 

C empty.html(содержимое которого было сгенерировано с помощью генератора): [https://github.com/atherdon/newsletters/blob/master/archive/Thematic/2021/January/flatfile8/empty.html](https://github.com/atherdon/newsletters/blob/master/archive/Thematic/2021/January/flatfile8/empty.html) 

В файле empty.html нет ни <head> ни <body>. также нету хеадера и футера, в которых у нас выводится спонсорские блоки.
Нашей задачей будет добить генерацию до состояния, когда контент из [content.md](http://content.md) будет сразу генерировать нам файл [https://github.com/atherdon/newsletters/blob/master/archive/Thematic/2021/January/flatfile8/template.html](https://github.com/atherdon/newsletters/blob/master/archive/Thematic/2021/January/flatfile8/template.html)

у нас есть нормальный test coverage - мы постарались проверить кучу cases - которые я смог словить, когда мы делали реальные newsletters с разным контентом для наших клиентов.

для генерации верстки для контента я использую `yarn parse`

для тестирования полного генератора я использую `yarn parse:full`

---

### 3. telegram-bot

еще была идея сделать телеграм бота, который бы парсил rss и отправлял бы ссылки подписчикам. чтото типо email newsletter только автоматическое и по интересам.

**Reply by atherdon** [reply]
@kolserdav нужно ли нам попробывать параллельно обсуждать другие проекты - или это только будет вас отвлекать?

Как я вижу это: например мы можем делать бекенд как основной проект, а пока мы обсуждаем новый майлстоун - можно сделать таски например в генераторе.

но я буду полностью отталкиваться от вас - как скажете - так и будет

**Reply by kolserdav** [reply]
Наверное обсуждать нужно. Чтобы мне видеть картину шире. А если будет пока не до этого, просто повременю с ответами.

**Reply by atherdon** [reply]
да, я как раз об этом и думал. я просто запишу несколько видео, а дальше мы уже будем смотреть - делать не делать и т.д.

**Reply by kolserdav** [reply]
Не помню, говорил или нет. По видео мне очень неудобно. Я их могу посмотреть для общего развития, но для задания только текст воспринимаю. То есть если окажется, что в видео о чем-то говорилось, но не было продублировано текстом, то сразу говорю для получения заданий я всегда настаиваю на текстовом описании. Из видео нельзя копировать и чтобы найти что-то конкретное спустя какое то время, уходит много времени.

**Reply by kolserdav** [reply]
Когда готовишь задание к сдаче, бывает по нескольку раз просматриваешь все пункты, чтобы ничего не упустить. А смотреть одно и то же кино по нескольку раз это пустая трата времени.

**Reply by atherdon** [reply]
Я вообще только за чтобы делать текст и описывать все словами. Первая причина почему я начал записывать видео - иногда мне сложно подготавливать детальные описания и т.д. Поэтому проще прыгнуть в Loom - записать видео и показать то что я имею ввиду. Но я понял - буду описывать текстом. нету никаких проблем - просто потребуется чуть больше времени на подготовку.

**Reply by atherdon** [reply]
@kolserdav я немного обновил информацию о markdown-generator, но мне кажется что мне все равно нужно записать видео и показать как все работает в реале :)

**Reply by kolserdav** [reply]
Записывайте, раз нужно)
